### ISSUE TITLE
Exclusions Crashing App

### ISSUE THREAD

**Issue by Trasd** [d4rken/sdmaid-public#274]
SD_Maid-v4.0.2(4002)-BETA-bdcf397.apk is crashing whenever I tap Exclusions.  Thereafter, it crashes every time I run it until I clear data.

Samsung Galaxy TabPRO SM-T900 12.2" with Android 5.1.1, unrooted.

Log file emailed.  I uninstalled and reinstalled, same results.

If you need more info, let me know.

**Comment by Trasd** [comment]
Slightly off topic: why can I not assign a priority or tags, etc.?

**Comment by d4rken** [comment]
I know where it crashes but can't seem to be able to reproduce it.
Do you have exclusions added ?

> Slightly off topic: why can I not assign a priority or tags, etc.?
Github handles this differently than Bitbucket. It's up to the maintainer to label the issues.

**Comment by Spire42** [comment]
I can reproduce this 100% reliably with my rooted Nexus 6P running Android 6.0.1. It happens whether or not there are any exclusions defined — even on a completely fresh install.

**Comment by Trasd** [comment]
No, I cannot even get to exclusions.  I cleared data and also did an uninstall/reinstall, both which should have cleared any exclusions I had.
Troy
On Jan 1, 2016 10:25 AM, Matthias Urhahn <notifications@github.com> wrote:I know where it crashes but can't seem to be able to reproduce it.
Do you have exclusions added ?


Slightly off topic: why can I not assign a priority or tags, etc.?
Github handles this differently than Bitbucket. It's up to the maintainer to label the issues.

**Comment by d4rken** [comment]
Can reproduce it it now, seems to be a proguard issue.

**Comment by Trasd** [comment]
Forgot to add a method to your proguard exception list? LOL, I ran round and round with a method that worked fine compiled with debug on, but when I compiled the release version, it wouldn't run.

I had no idea proguard's obfuscation made some methods fail as my IDE setup defaults and I had not even begun to look into PG options.  I was completely unaware.

Luckily, you're not unaware and I'm glad you found the problem!  Thanks darken.

**Comment by d4rken** [comment]
Because it is part of the support library i was under the impression that the necessariy proguard rules were already specified in the library, apparently not.

Specifically it was this: https://code.google.com/p/android/issues/detail?id=58508
I don't really agree that this is a user(dev) error.

**Comment by Trasd** [comment]
Yes, I agree; this is not user error.  What makes it worse for new Android developers is that some of us have been programming for years and are writing programs a bit more complicated than "Hello World."  I've been programming for years in Assembly and C++.

I hadn't the slightest idea proguard could screw things up like this until I happened to be on the site looking for other non-related information.  I didn't even know what proguard was when I ran into the error!  That made it nearly impossible for me to fix the problem and I almost shelved the project.

If it gives you trouble, I'm wondering if there is any hope for me!  LOL.
